src/mono/netcore/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventPipe.Mono.cs

```diff
@@ -0,0 +1,93 @@
+                    handle.SafeWaitHandle.DangerousRelease();
+            }
+        }
+
+        internal static void WaitOne(EventPipeWaitHandle handle)
+        {
+            Debug.Assert(!handle.SafeWaitHandle.IsInvalid);
+            handle.WaitOne();
+        }
+
+        // These ICalls are used by the configuration APIs to interact with EventPipe.
+        //
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        private static unsafe extern ulong Enable(char* outputFile, EventPipeSerializationFormat format, uint circularBufferSizeInMB, EventPipeProviderConfigurationNative* providers, uint numProviders);
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern void Disable(ulong sessionID);
+
+        //
+        // These ICalls are used by EventSource to interact with the EventPipe.
+        //
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern IntPtr CreateProvider(string providerName, Interop.Advapi32.EtwEnableCallback callbackFunc);
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern unsafe IntPtr DefineEvent(IntPtr provHandle, uint eventID, long keywords, uint eventVersion, uint level, byte* pMetadata, uint metadataLength);
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern unsafe IntPtr GetProvider(char* providerName);
+
+        internal static unsafe IntPtr GetProvider(string providerName)
+        {
+            fixed (char* p = providerName)
+            {
+                return GetProvider(p);
+            }
+        }
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern void DeleteProvider(IntPtr provHandle);
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern int EventActivityIdControl(uint controlCode, ref Guid activityId);
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern unsafe void WriteEventData(IntPtr eventHandle, EventProvider.EventData* pEventData, uint dataCount, Guid* activityId, Guid* relatedActivityId);
+
+
+        //
+        // These ICalls are used as part of the EventPipeEventDispatcher.
+        //
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern unsafe bool GetSessionInfo(ulong sessionID, EventPipeSessionInfo* pSessionInfo);
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern unsafe bool GetNextEvent(ulong sessionID, EventPipeEventInstanceData* pInstance);
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern unsafe IntPtr GetWaitHandle(ulong sessionID);
+
+        // ICall used for cross platform event signaling.
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        private static extern bool SetEventInternal(IntPtr handle);
```

Comment:
no corresponding definition in `icall-def-netcore.h`

Reply:
Currently not called, so maybe I just shortcut this in managed layer waiting to add potential icalls when I start to implement the native code for EventPipe.

Reply:
You can just use [`EventWaitHandle.Set` internal method](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Threading/EventWaitHandle.Windows.cs#L92). It already exists on all platforms.

Reply:
Will do.